Mathlib/Analysis/Distribution/ContDiffMapSupportedIn.lean

```diff
@@ -0,0 +1,121 @@
+/-
+Copyright (c) 2023 Anatole Dedecker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anatole Dedecker, Luigi Massacci
+-/
+
+import Mathlib.Analysis.Calculus.ContDiff.Defs
+import Mathlib.Topology.ContinuousMap.Bounded.Normed
+import Mathlib.Topology.Sets.Compacts
+
+/-!
+# Continuously differentiable functions supported in a given compact
+
+This file develops the basic theory of bundled `n`-times continuously differentiable functions
+with support contained in a given compact.
+
+Given `n : ℕ∞` and a compact subset `K` of a normed space `E`, we consider the type of bundled
+functions `f : E → F` (where `F` is a normed vector space) such that:
+
+- `f` is `n`-times continuously differentiable: `ContDiff ℝ n f`.
+- `f` vanishes outside of a compact: `EqOn f 0 Kᶜ`.
+
+We define it as a bundled structure so as to endow it with the topology required to make it a ù
+subspace of the space of test functions.
```

Comment:
There are a few more things that would be interesting to add (sorry for not helping earlier with the docstring...):
```suggestion
The main reason this exists as a bundled type is to be endowed with its natural locally convex topology
(namely, uniform convergence of `f` and its derivative up to order `n`). Taking the locally convex inductive
limit of these as `K` yields the natural topology on test functions, used to define distributions.

While most of distribution theory cares only about `C^∞` functions, we also want to endow the space of `C^n`
test functions with its natural topology. Indeed, distributions of order less than `n` are precisely those
which extend continuously to this larger space of test functions.
```